cobalt/black_box_tests/testdata/service_worker_message_test.html

```diff
@@ -116,39 +126,14 @@
             registration.active.postMessage({ message:
                 'Registration ready resolved.'});
             count_event(1);
-            window.setTimeout(
-                () => {
-                    registration.unregister()
-                        .then(function (success) {
-                            console.log('(Expected) unregister success :',
-                                success);
-                            count_event(9);
-                        }, function (error) {
-                            console.log('(Unexpected) unregister ' +
-                                `${error}`, error);
-                            assertIncludes('SecurityError: ', `${error}`);
-                            notReached();
-                        });
-                }, 1000);
+            window.activeServiceWorker = registration.active;
         });
 
         navigator.serviceWorker.register('service_worker_message_test.js', {
             scope: './',
-        }).catch(function (error) {
-            console.log('(Unexpected) :', error);
-            notReached();
-        });
+        }).catch(notReached);
 
         setupFinished();
-        // This delay has to be long enough to guarantee that the test has
-        // finished.
-        window.setTimeout(
-            () => {
-                console.log('Events:', expected_event_count)
-                assertEqual(9, expected_event_count);
```

Comment:
If you remove this, the test won't check it all the expected points have been reached.

Reply:
Discussed offline. The count(9) would assert if any are skipped, and the test wouldn't end if that one is skipped.